### PR TITLE
adjustment the onResize trigger time in order to get correct Safe Are…

### PR DESCRIPTION
**Pull request by knoxHuang**
…a in iOS

https://github.com/cocos-creator/2d-tasks/issues/2503

### CODE REVIEW

File: engine/jsb-game.js [hunk 1]
```diff
@@ -35,12 +35,28 @@ jsb.onResume = function () {
...
     size.height /= cc.view._devicePixelRatio;
     window.resize(size.width, size.height);
     cc.view.setCanvasSize(window.innerWidth, window.innerHeight);
+}
+
+jsb.onResize = function (size) {
+    if (size.width === 0 || size.height === 0) return;
+
+    // getSafeAreaEdge is asynchronous in iOS, so delayed adaptation is required
+    if (cc.sys.isNative && cc.sys.os === cc.sys.OS_IOS) {
```
Comment (jareguo) [inline engine/jsb-game.js]: on iOS
delayed adaptation -> callback later
cc.sys.isNative -> CC_JSB

File: engine/jsb-game.js [hunk 2]
```diff
@@ -35,12 +35,28 @@ jsb.onResume = function () {
...
+        let edge = jsb.Device.getSafeAreaEdge();
+        let useSafeArea = !(edge.x === 0 && edge.y === 0 && edge.z === 0 && edge.w === 0);
```
Comment (jareguo) [inline engine/jsb-game.js]: let hasSafeArea = (edges.x > 0 || edges.y > 0 || edges.z > 0 || edges.w > 0);

File: engine/jsb-game.js [hunk 3]
```diff
@@ -35,12 +35,28 @@ jsb.onResume = function () {
...
+        if (useSafeArea) {
+            setTimeout(() => {
+                resize(size);
```
Comment (jareguo) [inline engine/jsb-game.js]:
```
setTimeout(() => {
    if (cc.Vec4.strictEquals(edges, jsb.Device.getSafeAreaEdge())) {
        setTimeout(resize, 200, size);
    }
    else {
        resize(size);
    }
}, 0);
```

File: engine/jsb-game.js [hunk 4]
```diff
@@ -35,12 +35,30 @@ jsb.onResume = function () {
...
     cc.view.setCanvasSize(window.innerWidth, window.innerHeight);
+}
+
+jsb.onResize = function (size) {
+    if (size.width === 0 || size.height === 0) return;
+
+    // getSafeAreaEdge is asynchronous on iOS, so callback later is required
+    if (cc.sys.isNative && cc.sys.os === cc.sys.OS_IOS) {
+        let edge = jsb.Device.getSafeAreaEdge();
+        let hasSafeArea = (edges.x > 0 || edges.y > 0 || edges.z > 0 || edges.w > 0);
```
Comment (jareguo) [inline engine/jsb-game.js]: 变量未定义

File: engine/jsb-game.js [hunk 5]
```diff
@@ -35,12 +35,30 @@ jsb.onResume = function () {
...
+
+jsb.onResize = function (size) {
+    if (size.width === 0 || size.height === 0) return;
+
+    // getSafeAreaEdge is asynchronous on iOS, so callback later is required
+    if (cc.sys.isNative && cc.sys.os === cc.sys.OS_IOS) {
+        let edges = jsb.Device.getSafeAreaEdge();
+        let hasSafeArea = (edges.x > 0 || edges.y > 0 || edges.z > 0 || edges.w > 0);
+        if (hasSafeArea) {
+            if (cc.Vec4.strictEquals(edges, jsb.Device.getSafeAreaEdge())) {
```
Comment (umbrellaPP) [inline engine/jsb-game.js]: 看注释的意思应该是延时 getSafeAreaEdge，而不是 延时 resize

**Pull request merged**